aya/src/programs/mod.rs

```diff
@@ -410,6 +410,39 @@ impl Program {
             Self::CgroupDevice(p) => p.fd(),
         }
     }
+
+    /// Returns information about a loaded program with the [`ProgramInfo`] structure.
+    ///
+    /// This information is populated at load time by the kernel and can be used
+    /// to get kernel details for a given [`Program`].
+    pub fn program_info(&self) -> Result<ProgramInfo, ProgramError> {
+        match self {
+            Self::KProbe(p) => p.program_info(),
```

Comment:
This pattern (match on every variant, pull out the MapData, call a function on the MapData) repeats at least 4 times in this file. Can you extract a private function that pulls out the MapData and use it in all these functions? Ideally, that would done in a parent comment.

Reply:
This is not using MapData, wrong file :P The inner types are actually different, so I think you could only do it with a macro.

Reply:
Ah, thanks for explaining.